### DISCUSSION TITLE
Feature Request: Autocomplete on tag input

### DISCUSSION THREAD

**Discussion by eoinmurray** [segmentio/evergreen#1260] [Ideas]
I tried making this myself, but it didn't work out well. Autocomplete on tags would be very useful since most sites that accept tags use autocomplete for the tag input (stackoverflow etc...).

Maybe the api could be like the following:

```
<Autocomplete
  title="Fruits"
  onChange={(newTags) => console.log(newTags)}
  items={['Apple', 'Apricot', 'Banana', 'Cherry', 'Cucumber']}
>
  {(props) => {
    const { getInputProps, getRef, inputValues } = props
    return (
      <TagInput
        inputProps={{ placeholder: 'Add trees...' }}
        values={inputValues}
        innerRef={getRef}
        {...getInputProps()}
      />
    )
  }}
</Autocomplete>
```

**Comment by mshwery** [comment]
We have implemented something like this before, but it was fully custom (wasn't using `TagInput`).

What do you think about allowing consumers to pass in their own `inputComponent`?

**Comment by archived-m** [comment]
I'm also looking for something like this.

It would be cool if it functioned like `AutoComplete`, passing a prop like `suggestions` which is also just an array of strings.

If I find the time somewhere next week, I'll try it out and submit a PR

**Comment by eoinmurray** [comment]
Yeah even making it more modular so it takes any type of inputComponent and lets the user prepare the suggestions would work.

We've had to use react-select for our tags input, and it nice - but it would be great to use evergreen for everything for visual homogeneity.
